__tests__/atomWithHash_spec.tsx

```diff
@@ -133,6 +133,74 @@ describe('atomWithHash', () => {
       expect(window.location.search).toEqual('?q=foo');
       expect(window.location.hash).toEqual('#count=2');
     });
+    window.history.back();
```

Comment:
i also added this to the other replace state test, because i think as written it doesn't actually check that the replace worked. Now it checks that the initial replace causes the history not to have a rogue `/?q=foo`.

 I could be wrong but i think before this test just checked that the browser history works normally since it was manually pushing "another" on. (Please correct me if I'm wrong here)

Reply:
I took a look at it and it seems like you are right. I like the addition to this test, and it feels more thorough.